transpose_comments.py

```diff
@@ -86,63 +88,98 @@ def main_transpose_comments():
     templates = [os.path.join(template_dir, filename) for filename in os.listdir(template_dir) if filename.endswith('docx')]
     modified_docx = [os.path.join(modified_docx_dir, filename) for filename in os.listdir(modified_docx_dir) if filename.endswith('docx')]
     template2modified_docx = map_templates_to_modified_reports(templates, modified_docx)
+    print(template2modified_docx)
     transpose_modification_to_new_reports(template2modified_docx)
     assert len(os.listdir(transposed_docx_dir)) == 110
     shutil.rmtree(os.path.join("reports_word", "temp_transposition"))
 
 # Fonction nécessaire
 
-def mkdir_ifnotexist(path) :
-    if not os.path.isdir(path) :
+def mkdir_ifnotexist(path: str):
+    """
+    Create a folder if it's doesn't exist
+    """
+    if not os.path.isdir(path):
         os.mkdir(path)
         
 
-def encode_name(name):
-    # Normalise le nom de la mesure ou volet, notamment pour l'utiliser comme nom de code dans les commentaires
+def normalize_name(name: str) -> str:
+    """
+    Normalize a str: delete whitespace, special characters, put in lowercase and keep only letters
+    """
     name = name.lower()
     name = unidecode(name)
     name = re.sub('[^a-z]', ' ',  name)
     name = re.sub(' +', '', name)
     return name
 
-def flatten(L):
-    # Aplatir une liste imbriquée [[., ., [[.]]]] -> [., ., .]
+def flatten(L: list):
+    """
+    Transform list like: [[., ., [[.]]]] into a classical structure list: [., ., .]
+    """
     if type(L) is list:
         for item in L:
             yield from flatten(item)
     else:
         yield L
         
 
-def gen_unit_list(L):
-    # Générer 
+def gen_unit_list(L: list):
+    """
+    ???
```

Comment:
Creates a generator that generates the smallest list for nested lists : [ [[.], [., .], [.]], [ [a], [.] ] ] --> [.], [., .], [.], [a], [.]